### PR TITLE
Update uMurmur to 0.2.17 and for DSM 5+6 generic service use

### PR DESCRIPTION
**Pull request by m4tt075**
_Motivation:_ Initially, just to update `uMurmur` package for DSM 5+6 generic service use. While at it, I modernized `uMurmur` and dependencies too. 

_Linked issues:_ #3041 , #2904 

_Updates:_ 
- `uMurmur` to version 0.2.17 (latest release) and switch to CMAKE
- Default configuration file (removing privilege dropping and minor changes, in line with source default configuration)
- Cross-package `libconfig` to 1.7.2 (latest release)
- Cross-package  `protobuf-c` to 1.3.0 (latest release)

_Notes:_ Currently untested, as I neither know nor need this package. 

### Checklist
- [ ] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

File: spk/umurmur/src/gencert.sh [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
-    touch ${INSTALL_DIR}/var/umurmur.key
-    touch ${INSTALL_DIR}/var/umurmur.crt
+    echo "Certificate generation for uMurmur failed" >> ${INST_LOG}
+    touch ${SYNOPKG_PKGDEST}/var/umurmur.key
+    touch ${SYNOPKG_PKGDEST}/var/umurmur.crt
     exit 1
 fi
 
-echo "uMurmur's certificate successfully generated"
+echo "uMurmur's certificate successfully generated" >> ${INST_LOG}
```
Comment (ymartin59) [inline spk/umurmur/src/gencert.sh]: I support script is expected to be also started out of installer `postinst`, so I would remove redirection to `INST_LOG` that may not be defined.

File: spk/umurmur/src/service-setup.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+UMURMUR="${SYNOPKG_PKGDEST}/bin/umurmurd"
+CFG_FILE="${SYNOPKG_PKGDEST}/var/umurmur.conf"
+GEN_CERT="${SYNOPKG_PKGDEST}/sbin/gencert.sh"
+
+SERVICE_COMMAND="${UMURMUR} -c ${CFG_FILE} -p ${PID_FILE}"
+
+service_postinst ()
+{
+    # Certificate generation
+    ${GEN_CERT} > /dev/null 2>&1
```
Comment (ymartin59) [inline spk/umurmur/src/service-setup.sh]: Here you should redirect script output to `INST_LOG`

**Comment by ymartin59** [comment]
Is there a reason not to set `SERVICE_PORT` so that `app/config` and `umurmur.sc` are automatically generated ?

**Comment by m4tt075** [comment]
@ymartin59 The reason is that the udp protocol needs to be added, which is not done through the service script.

The port configuration has been flawed though, and I have therefore pushed some fixes. With those fixes I'm able to connect a Mumble client to the XPE 6.1 uMurmur server manually. That's a good sign. What is still not working is the service link though. I think(!) it should "just" open up the mumble client and connect it to the server (which works manually), but the link is not working (the cgi script is not found and therefore not executed). And I don't know how to make it work. From some old issues it seems the service link hasn't worked for some time. Any idea?

**Comment by ymartin59** [comment]
When reviewing I was wondering if cgi was in its right place in `app` folder... I would move to `bin` and probably nginx or apache requires additional settings to grant execution to that script

**Comment by ymartin59** [comment]
You were right to correct port: https://github.com/SynoCommunity/spksrc/wiki/SynoCommunity-Used-Ports

**Comment by ymartin59** [comment]
@m4tt075 May I remind you minor fixes I requested in review ?

File: spk/umurmur/src/gencert.sh (hunk 1 above)
Comment (ymartin59) [inline spk/umurmur/src/gencert.sh]: I suppose script is expected to be also started out of installer postinst, so I would remove redirection to INST_LOG that may not be defined.

**Comment by m4tt075** [comment]
@ymartin59 Sorry, drowning in RL these days. Lots of work. Lots of business travels. But it is not the minor fixes which are the problem. I have been trying to install and tweak the configurations of nginx and apache some time ago to make the service link work. Failed miserably, but I don't know whether this was due to the package or my inability to configure those progams appropriately. Anyways, got stuck, and not sure this is ready to merge, independent of your minor tweaks. Sorry again.

**Comment by ymartin59** [comment]
OK. Noticed.

**Comment by m4tt075** [comment]
@ymartin59 : I revisited this PR today and addressed your change requests. I still haven't figured out how to test the cgi script. Having said that, this PR is very unlikely to introduce any regressions either. I will create some test builds to offer it to the community. Certainly doesn't harm. And then we can still decide what to do with it. OK?

**Comment by m4tt075** [comment]
DSM 6.1 test builds all [green](https://travis-ci.org/m4tt075/spksrc/builds/377429205).
Test packages available [here](https://github.com/m4tt075/spksrc/releases/tag/umu-0.2.17).

**Comment by mikhnal** [comment]
Sadly there is no package for broadwellnk :(

**Comment by m4tt075** [comment]
@mikhnal Difficult to keep track with all the architectures Synolog keeps adding all the time...  :)
Compiled it for you --> [Here](https://www.dropbox.com/s/1edak9vknga4ucd/umurmur_broadwellnk-6.2_0.2.17-7.spk?dl=0). But please report back what you find, ok?

**Comment by m4tt075** [comment]
@mikhnal Did it work for you?

**Comment by mikhnal** [comment]
No, this version didn't work. In logs I see:
`May 14 08:05:55 INFO: TERM signal. Shutting down.
May 14 08:06:01 INFO: PID-file: /volume1/@appstore/umurmur/var/umurmur.pid
May 14 08:06:02 INFO: OpenSSL library initialized
May 14 08:06:02 INFO: No channel links found in configuration file.
May 14 08:06:02 FATAL: initgroups() failed: Operation not permitted`
Version 0.2.16 works like a harm

**Comment by m4tt075** [comment]
OK. Thanks, @mikhnal .

Could you please post the full `umurmur.log` and `umurmur_install.log` here? You will find them in `/usr/local/umurmur/var` once you `ssh` into your NAS box.

**Comment by mikhnal** [comment]
Ok, here go. 
`umurmur_install.log`
```
Thu May 10 22:49:05 CEST 2018
===> Step preupgrade. USER=umurmur GROUP= SHARE_PATH=
Thu May 10 22:49:09 CEST 2018
===> Step preinst. USER=umurmur GROUP= SHARE_PATH=
Thu May 10 22:49:11 CEST 2018
===> Step postinst. USER=umurmur GROUP= SHARE_PATH=
Invoke service_postinst
uMurmur's certificate successfully generated
Granting 'sc-umurmur' unix ownership on /volume1/@appstore/umurmur/var
Thu May 10 22:49:11 CEST 2018
===> Step postupgrade. USER=umurmur GROUP= SHARE_PATH=
Granting 'sc-umurmur' unix ownership on /volume1/@appstore/umurmur/var
Thu May 10 23:18:34 CEST 2018
===> Step preuninst. USER=umurmur GROUP= SHARE_PATH=
Thu May 10 23:18:34 CEST 2018
===> Step postuninst. USER=umurmur GROUP= SHARE_PATH=
Removing user sc-umurmur
Thu May 10 23:21:40 CEST 2018
===> Step preinst. USER=umurmur GROUP= SHARE_PATH=
Thu May 10 23:21:42 CEST 2018
===> Step postinst. USER=umurmur GROUP= SHARE_PATH=
Invoke service_postinst
uMurmur's certificate successfully generated
Granting 'sc-umurmur' unix ownership on /volume1/@appstore/umurmur/var
Fri May 11 00:22:40 CEST 2018
===> Step preuninst. USER=umurmur GROUP= SHARE_PATH=
Fri May 11 00:22:40 CEST 2018
===> Step postuninst. USER=umurmur GROUP= SHARE_PATH=
Removing user sc-umurmur
Mon May 14 08:05:56 CEST 2018
===> Step preupgrade. USER=umurmur GROUP= SHARE_PATH=
Mon May 14 08:05:58 CEST 2018
===> Step preinst. USER=umurmur GROUP= SHARE_PATH=
Mon May 14 08:05:58 CEST 2018
===> Step postinst. USER=umurmur GROUP= SHARE_PATH=
Invoke service_postinst
uMurmur's certificate successfully generated
Granting 'sc-umurmur' unix ownership on /volume1/@appstore/umurmur/var
Mon May 14 08:05:58 CEST 2018
===> Step postupgrade. USER=umurmur GROUP= SHARE_PATH=
Granting 'sc-umurmur' unix ownership on /volume1/@appstore/umurmur/var 
```

`umurmur.log`
```
Mon May 14 09:05:15 CEST 2018
Starting umurmur command /volume1/@appstore/umurmur/bin/umurmurd -c /volume1/@appstore/umurmur/var/umurmur.conf -p /volume1/@appstore/umurmur/var/umurmur.pid 
```

`umurmurd.log` without my personal data
```
May 11 00:23:20 INFO: PID-file: /usr/local/umurmur/var/umurmur.pid
May 11 00:23:20 INFO: Switch to user 'nobody' group 'nobody'
May 11 00:23:20 INFO: OpenSSL library initialized
May 11 00:23:20 INFO: No channel links found in configuration file.
May 11 00:23:20 INFO: uMurmur version 0.2.16 ('Agrajag') protocol version 1.2.4
May 11 00:23:20 INFO: Visit http://code.google.com/p/umurmur/
May 11 00:23:33 WARN: SSL: verify error:num=18:self signed certificate:depth=0:/CN=Mumble User
May 11 00:23:33 INFO: Connection closed by peer - [1] @x.x.x.x:51088
May 11 00:23:35 WARN: SSL: verify error:num=18:self signed certificate:depth=0:/CN=Mumble User
May 11 00:23:35 INFO: User mikhnal authenticated - [1] u@x.x.x.x:51089
May 11 00:23:35 INFO: New UDP connection from x.x.x.x on port 58204 - [1] u@x.x.x.x:51089
May 11 00:25:40 INFO: Connection closed by peer - [1] u@x.x.x.x:51089
May 11 07:09:03 WARN: SSL: verify error:num=18:self signed certificate:depth=0:/CN=U\xC3\x85\xC2\xBCytkownik Mumble
May 11 07:09:03 INFO: Connection closed by peer - [1] @x.x.x.x:54129
May 11 07:09:06 WARN: SSL: verify error:num=18:self signed certificate:depth=0:/CN=U\xC3\x85\xC2\xBCytkownik Mumble
May 11 07:09:06 INFO: User x authenticated - [1] u@x.x.x.x:54130
May 11 07:09:06 INFO: New UDP connection from x.x.x.x on port 58607 - [1] u@x.x.x.x:54130
May 11 07:42:25 INFO: Connection closed by peer - [1] u@x.x.x.x:54130
May 14 08:05:55 INFO: TERM signal. Shutting down.
May 14 08:06:01 INFO: PID-file: /volume1/@appstore/umurmur/var/umurmur.pid
May 14 08:06:02 INFO: OpenSSL library initialized
May 14 08:06:02 INFO: No channel links found in configuration file.
May 14 08:06:02 FATAL: initgroups() failed: Operation not permitted
May 14 08:07:16 INFO: PID-file: /volume1/@appstore/umurmur/var/umurmur.pid
May 14 08:07:16 INFO: OpenSSL library initialized
May 14 08:07:16 INFO: No channel links found in configuration file.
May 14 08:07:16 FATAL: initgroups() failed: Operation not permitted
May 14 09:05:15 INFO: PID-file: /volume1/@appstore/umurmur/var/umurmur.pid
May 14 09:05:15 INFO: OpenSSL library initialized
May 14 09:05:15 INFO: No channel links found in configuration file.
May 14 09:05:15 FATAL: initgroups() failed: Operation not permitted
```

**Comment by m4tt075** [comment]
Hmh, the install log looks fine. From the umurmur.log it seems umurmur is at least running. The umurmurd.log confuses me as it states that 4.2.16 is running. That is the old version. The new one is 4.2.17. The umurmurd.log file should not be used anymore. But this might be related to your configuration file, which has probably survived the upgrade. Coud you please post the umurmur.conf file after you have upgraded to 4.2.17 (removing all personal information again)?

**Comment by mikhnal** [comment]
This is my `umurmur.conf` file:
```
max_bandwidth = 128000;
welcometext = "Witaj na moim uMurku!";
certificate = "/usr/local/umurmur/var/umurmur.crt";
private_key = "/usr/local/umurmur/var/umurmur.key";
password = "xxx";
admin_password = "xxx";   # Set to enable admin functionality.
# ban_length = 0;            # Length in seconds for a ban. Default is 0. 0 = forever.
# enable_ban = false;        # Default is false
# banfile = "/usr/local/umurmur/var/banfile.txt";   # File to save bans to. Default is to not save bans to file.
# sync_banfile = false;      # Keep banfile synced. Default is false, which means it is saved to at shutdown only.
# allow_textmessage = true;  # Default is true
# opus_threshold = 100;      # Percentage of users supporting Opus codec for it to be chosen. Default is 100.
max_users = 4;

bindport = 60555;
# bindaddr = "192.168.1.1";

# username and groupname for privilege dropping.
# Will attempt to switch user if set. 
username = "nobody";
# If groupname not set the user's default login group will be used
groupname = "nobody";

# Log to file option. Default is logging to syslog.
# umurmurd will close and reopen the logfile if SIGHUP is received.
logfile = "/usr/local/umurmur/var/umurmurd.log";

# Channel tree definition:
# Root channel must always be defined first.
# If a channel has a parent, the parent must be defined before the child channel(s).
channels = ( {
	 name = "uMurek";
	 parent = "";
	 description = "Rodzinny uMurek";
#	 noenter = true;
#	 },
#	 {
#	 name = "Lobby";
#	 parent = "Root";
#	 description = "Lobby channel";
#	 },
#	 {
#	 name = "Red team";
#	 parent = "Lobby";
#	 description = "The Red team channel";
#	 # password = "redpw";
#	 },
#	 {
#	 name = "Blue team";
#	 parent = "Lobby";
#	 description = "The Blue team channel";
#	 # password = "bluepw";
	 }
);
# Channel links configuration.
#channel_links = ( {
#	 source = "Lobby";
#	 destination = "Red team";
#	 },
#	 {
#	 source = "Lobby";
#	 destination = "Blue team";
#	 }
#);

# The channel in which users will appear in when connecting.
# Note that default channel can't have 'noenter = true' or password set
default_channel = "uMurek";
```

**Comment by mikhnal** [comment]
I don't belive... It's works! I comment this two lines and this version run.

TYVM!

**Comment by m4tt075** [comment]
Very happy to help. I will have to include this replacement in the upgrade logic of the script. It is difficult to test this package as there are not a lot of test users out there. Based on the downloads, it is 3 so far, including you. Therefore, thanks a lot for reporting!  @ymartin59 : FYI.

**Comment by m4tt075** [comment]
@ymartin59 Just pushed upgrade scripts to fix the `umurmur.conf` files w.r.t.
- Changing log file location to default (allowing accessibility via the DSM package manager)
- Commenting out username and groupname `nobody` as this prevents umurmur from running on DSM6 systems (see user report above)

If the code looks good to you, I'm not really sure it is worthwhile to create more test packages as this package does not face a lot of demand (only 13 downloads within almost two weeks). But please take a look and judge for yourself...

**Comment by mikhnal** [comment]
You are my hero 🥇

**Pull request merged**